com.unity.render-pipelines.universal/Editor/Camera/CameraStackReorderableList.cs

```diff
@@ -0,0 +1,301 @@
+            }
+
+            if (!cameraList.Contains(cam))
+                cameraList.Add(cam);
+
+            return false;
+        }
+
+        #endregion
+
+        public event Action<Camera> OnCameraAdded;
+
+        public CameraStackReorderableList(Camera camera, CameraEditor.Settings settings, UniversalRenderPipelineSerializedCamera serializedCamera)
+        {
+            this.serializedCamera = serializedCamera;
+            this.camera = camera;
+            this.settings = settings;
+
+            m_LayerList = new ReorderableList(serializedCamera.serializedObject, serializedCamera.cameras, true, true, true, true)
+            {
+                drawHeaderCallback = rect => EditorGUI.LabelField(rect, Styles.cameras),
+                drawElementCallback = DrawElementCallback,
+                onSelectCallback = SelectElement,
+                onRemoveCallback = RemoveCamera,
+                onAddDropdownCallback = AddCameraToCameraList,
+                onCanRemoveCallback = list => list.count > 0
+            };
+        }
+
+        void RemoveCamera(ReorderableList list)
+        {
+            // As multi selection is disabled, selectedIndices will only return 1 element, remove that element from the list
+            if (list.selectedIndices.Any())
+            {
+                serializedCamera.cameras.DeleteArrayElementAtIndex(list.selectedIndices.First());
+            }
+            else
+            {
+                // Nothing selected, remove the last item on the list
+                ReorderableList.defaultBehaviours.DoRemoveButton(list);
+            }
+
+            serializedCamera.serializedObject.ApplyModifiedProperties();
+
+            ClearValidation();
+        }
+
+        void SelectElement(ReorderableList list)
+        {
+            var cam = serializedCamera[list.index];
+            if (cam == null)
+                return;
+
+            if (Event.current.clickCount == 2)
+            {
+                Selection.activeObject = cam;
+            }
+
+            EditorGUIUtility.PingObject(cam);
+        }
+
+        void DrawElementCallback(Rect rect, int index, bool isActive, bool isFocused)
+        {
+            rect.height = EditorGUIUtility.singleLineHeight;
+            rect.y += 1;
+
+            var element = serializedCamera.cameras.GetArrayElementAtIndex(index);
+
+            var cam = element.objectReferenceValue as Camera;
+            if (cam != null)
+            {
+                bool outputWarning = !Validate(cam, () => !IsStackCameraOutputDirty(cam), outputWarningCameras);
+
+                GUIContent nameContent =
+                    outputWarning ?
+                    EditorGUIUtility.TrTextContent(cam.name, "Output properties do not match base camera", CoreEditorStyles.warningIcon) :
+                    EditorGUIUtility.TrTextContent(cam.name);
+
+                var type = cam.gameObject.GetComponent<UniversalAdditionalCameraData>().renderType;
```

Comment:
is it possible to cache the result of cam.gameObject.GetComponent<UniversalAdditionalCameraData>() ? i am afraid we may call this function every frame and cause slowdowns

Reply:
We need to get an updated version of it to know the type of the camera on the list.